### PR TITLE
Not raise DeepchecksProcessError for UnderAnnotated + WeakSegments

### PR DESCRIPTION
**Pull request by harsh-deepchecks**
#### Reference Issues/PRs

Fixes [DEE-621 - WeakSegments + Under annotated checks return error if no segments found](https://linear.app/deepchecks/issue/DEE-621/weaksegments-under-annotated-checks-return-error-if-no-segments-found)

#### What does this implement/fix? Explain your changes.
The WeakSegments and UnderAnnotated checks earlier returns DeepchecksProcessError. But now, after the change, it will return a empty CheckResult in the following cases:
- If annotation ratio of the data is > 95%. (Only for UnderAnnotated check). This is a configurable parameter accepted by property and metadata checks.
- If weak segments is empty, it will return a message to try increasing the segment_by (either properties or metadata).

Also, both checks return a NotEnoughSamplesError if there are less than 10 samples. 

Below is the output for data integrity suite when Under annotated checks are skipped based on the above mentioned conditions: 
image

Running stand-alone under annotated check: 
image

### CODE REVIEW

File: deepchecks/nlp/checks/model_evaluation/weak_segments_performance.py [hunk 1]
```diff
@@ -105,8 +110,9 @@ def run_logic(self, context: Context, dataset_kind) -> CheckResult:
                                                    dummy_model=dummy_model, scorer=scorer)
 
         if len(weak_segments) == 0:
-            raise DeepchecksProcessError('WeakSegmentsPerformance was unable to train an error model to find weak '
-                                         f'segments. Try increasing n_samples or supply more {self.segment_by}.')
+            display_msg = 'WeakSegmentsPerformance was unable to train an error model to find weak segments.'\
+                          f'Try supplying more {self.segment_by}.'
```
Comment (Nadav-Barak) [inline deepchecks/nlp/checks/model_evaluation/weak_segments_performance.py:114]:
```suggestion
                          f'Try supplying additional {self.segment_by}.'
```

File: deepchecks/nlp/checks/model_evaluation/weak_segments_performance.py [hunk 2]
```diff
@@ -105,8 +110,9 @@ def run_logic(self, context: Context, dataset_kind) -> CheckResult:
...
+            return CheckResult(value={'message': display_msg}, display=[display_msg])
```
Comment (Nadav-Barak) [inline deepchecks/nlp/checks/model_evaluation/weak_segments_performance.py:115]: can have the value directly a str I think

File: deepchecks/nlp/checks/data_integrity/under_annotated_segments.py [hunk 3]
```diff
@@ -30,15 +30,20 @@
...
 
 class UnderAnnotatedSegments(SingleDatasetCheck, WeakSegmentAbstract):
     """Check for under annotated data segments."""
 
     def __init__(self, segment_by: str, columns: Union[Hashable, List[Hashable], None],
                  ignore_columns: Union[Hashable, List[Hashable], None], n_top_features: int,
-                 segment_minimum_size_ratio: float, n_samples: int,
-                 categorical_aggregation_threshold: float, n_to_show: int, **kwargs):
+                 segment_minimum_size_ratio: float, n_samples: int,  n_to_show: int,
+                 categorical_aggregation_threshold: float, annotation_ratio_threshold: float,
```
Comment (Nadav-Barak) [inline deepchecks/nlp/checks/data_integrity/under_annotated_segments.py:45]: annotation_ratio_threshold should be a float

File: deepchecks/nlp/checks/data_integrity/under_annotated_segments.py [hunk 4]
```diff
@@ -232,6 +249,8 @@ class UnderAnnotatedPropertySegments(UnderAnnotatedSegments):
         Maximum number of samples to use for this check.
     n_to_show : int , default: 3
         number of segments with the weakest performance to show.
+    annotation_ratio_threshold : float , default: 95.0
```
Comment (Nadav-Barak) [inline deepchecks/nlp/checks/data_integrity/under_annotated_segments.py:252]: I think we should not make it a argument but a const in the class

File: deepchecks/nlp/checks/model_evaluation/weak_segments_performance.py [hunk 5]
```diff
@@ -55,6 +57,9 @@ def run_logic(self, context: Context, dataset_kind) -> CheckResult:
         text_data = context.get_data_by_kind(dataset_kind)
         text_data = text_data.sample(self.n_samples, random_state=context.random_state, drop_na_label=True)
 
+        if text_data.n_samples < MIN_TEXT_SAMPLES:
+            raise NotEnoughSamplesError(f'Not enough samples to find weak {self.segment_by} segments.'
+                                        ' Minimum 10 samples required.')
```
Comment (Nadav-Barak) [inline deepchecks/nlp/checks/model_evaluation/weak_segments_performance.py:62]:
```suggestion
                                        f' Minimum {MIN_TEXT_SAMPLES} samples required.')
```

**Review by Nadav-Barak — APPROVED** [review]
Approved with small comments

File: deepchecks/nlp/checks/model_evaluation/weak_segments_performance.py (hunk 2 above)
Comment (harsh-deepchecks) [reply deepchecks/nlp/checks/model_evaluation/weak_segments_performance.py:115]: Actually the conditions expect the value to be a Dict.

File: deepchecks/nlp/checks/data_integrity/under_annotated_segments.py (hunk 4 above)
Comment (harsh-deepchecks) [reply deepchecks/nlp/checks/data_integrity/under_annotated_segments.py]: Done

File: deepchecks/nlp/checks/data_integrity/under_annotated_segments.py (hunk 3 above)
Comment (harsh-deepchecks) [reply deepchecks/nlp/checks/data_integrity/under_annotated_segments.py]: Made it a const instead of parameter

**Pull request merged**